src/packages/core/tests/server.test.ts

```diff
@@ -1257,58 +1261,287 @@ describe("server", () => {
-        const hash = s.provider.send("eth_sendTransaction", [{ from, data }]);
-        await s.provider.once("message");
+        const expectedResponse = Buffer.concat(chunks);
+
+        let sendCallCount = 0;
+        const send = (
+          message: Buffer,
+          _isBinary?: boolean,
+          _compress?: boolean
+        ) => {
+          sendCallCount++;
+          // the received message should not be chunked further or joined with
+          // anything else
+          assert.strictEqual(
+            message.toString("utf8"),
+            expectedResponse.toString("utf8")
+          );
+          return true;
+        };
+        const mockWebsocket: any = {
+          send,
+          cork: (callback: () => void) => callback(),
+          // if we are sending a single chunk `send` must be used instead of
+          // sendFirstFragment/sendFragment/sendLastFragment
+          sendFragment: () => assert.fail("should not have used sendFragment"),
+          sendFirstFragment: () =>
+            assert.fail("should not have used sendFirstFragment"),
+          sendLastFragment: () =>
+            assert.fail("should not have used sendLastFragment")
+        };
+        const dataGenerator = function* () {
+          for (const chunk of chunks) yield chunk;
+        };
+        sendFragmented(mockWebsocket, dataGenerator(), false, chunkSize);
+        assert.strictEqual(sendCallCount, 1, "send called too many times!");
+      });
 
-        // send a `debug_traceTransaction` request to the *server* so we can
-        // test for `transfer-encoding: chunked` and bufferfication.
-        const jsonRpcJson: any = {
-          jsonrpc: "2.0",
-          id: "1",
-          method: "debug_traceTransaction",
-          params: [await hash]
+      it("handles small+large+small bufferification edge-case", () => {
+        // the code path this triggers would be very difficult to trigger in an
+        // integration test, so we are testing the method directly instead.
+
+        const expectedResponses = [
+          // fits in a fragment
+          Buffer.from("hello", "utf8"),
+          // this second chunk is too large to fit in a single fragment (it is
+          // larger than bufferSize), so it should be sent by itself.
+          // Technically we _could_ copy parts of it into the previous fragment,
+          // but we haven't tested if this would be better or worse than sending
+          // this chunk as its own huge fragment.
+          Buffer.allocUnsafe(chunkSize + 1).fill(255),
+          // fits in a fragment
+          Buffer.from("world", "utf8")
+        ];
+        const dataGenerator = function* () {
+          for (const response of expectedResponses) yield response;
+        };
+        const receivedFragments = [];
+        const mockWebsocket: any = {
+          cork: (callback: () => void) => callback(),
+          sendFragment: (value: string) => receivedFragments.push(value),
```

Comment:
Should we distinguish between `sendFragment` / `sendFirstFragment` / `sendLastFragment`?

ie if `sendFragment` were just called 3 times, then this test would incorrectly pass.